forte/sparse_ci/determinant_substitution_lists.cc

```diff
@@ -129,276 +128,311 @@ void DeterminantSubstitutionLists::build_strings(const DeterminantHashVec& wfn)
                 a_add = it->second;
             }
             alpha_a_strings_.resize(naalpha);
-            alpha_a_strings_[a_add].push_back(std::make_pair(ii, I));
+            alpha_a_strings_[a_add].emplace_back(ii, I);
         }
     }
 }
 
 void DeterminantSubstitutionLists::op_s_lists(const DeterminantHashVec& wfn) {
     timer ops("Single sub. lists");
-
     if (!quiet_) {
         print_h2("Computing 1 Coupling Lists");
     }
+    lists_1a(wfn);
+    lists_1b(wfn);
+}
+
+void DeterminantSubstitutionLists::lists_1a(const DeterminantHashVec& wfn) {
+    timer ann("A lists");
 
-    // Get a reference to the determinants
     const det_hashvec& dets = wfn.wfn_hash();
+    std::map<size_t, size_t> vec_size;
 
-    timer ann("A lists");
-    for (size_t b = 0, max_b = beta_strings_.size(); b < max_b; ++b) {
+    for (auto& c_dets : beta_strings_) {
         size_t na_ann = 0;
         std::vector<std::vector<std::pair<size_t, short>>> tmp;
-        std::vector<size_t>& c_dets = beta_strings_[b];
         det_hash<int> map_a_ann;
-        for (size_t I = 0, maxI = c_dets.size(); I < maxI; ++I) {
-            size_t index = c_dets[I];
+        for (unsigned long index : c_dets) {
             const Determinant& detI = dets[index];
 
-            const std::vector<int>& aocc = detI.get_alfa_occ(ncmo_);
-            int noalfa = aocc.size();
+            const std::vector<int>& aocc = detI.get_alfa_occ(static_cast<int>(ncmo_));
 
-            for (int i = 0; i < noalfa; ++i) {
-                int ii = aocc[i];
+            for (int ii : aocc) {
                 Determinant detJ(detI);
                 detJ.set_alfa_bit(ii, false);
                 double sign = detI.slater_sign_a(ii);
                 size_t detJ_add;
                 auto search = map_a_ann.find(detJ);
                 if (search == map_a_ann.end()) {
                     detJ_add = na_ann;
-                    map_a_ann[detJ] = na_ann;
+                    map_a_ann[detJ] = static_cast<int>(na_ann);
                     na_ann++;
                     tmp.resize(na_ann);
                 } else {
                     detJ_add = search->second;
                 }
-                tmp[detJ_add].push_back(std::make_pair(index, sign > 0.0 ? (ii + 1) : (-ii - 1)));
+                tmp[detJ_add].emplace_back(index, sign > 0.0 ? (ii + 1) : (-ii - 1));
             }
         }
-        // size_t idx = 0;
-        for (auto& vec : tmp) {
-            // ignore (N-1)-electron det. only coupled to itself
-            if (vec.size() > 1) {
+
+        for (const auto& vec : tmp) {
+            // TODO: if (vec.size() > 1) { ??
```

Comment:
Here I am unsure if the correct way is `!vec.empty()` or `vec.size() > 1`. They actually correspond to checking for `>= 1` and `> 1`, so they are not equivalent. (question for @lcyyork).